scripts/build-upgrade-image.sh

```diff
@@ -62,15 +62,65 @@ for deb_tarball in "$LIVE_BUILD_OUTPUT_DIR/$APPLIANCE_VARIANT"*.debs.tar.gz; do
 	tar xf "$deb_tarball" -C debs
 done
 
+#
 # Generate an Aptly/APT repository
+#
 aptly repo create -distribution=bionic -component=delphix upgrade-repository
 aptly repo add upgrade-repository debs
 aptly publish repo -skip-contents -skip-signing upgrade-repository
 
+#
+# Generate the "payload" of the upgrade image, which will contain the
+# upgrade scripts, as well as the Aptly/APT repository. We generate this
+# "payload" tarball as a way to limit the number of files that we need
+# to checksum and store in "upgrade image" tarball itself. Rather than
+# the "upgrade image" tarball containing all of the files for the upgrade
+# scripts and APT repository, it'll only contain the "payload.tar.gz"
+# file, and a few other miscellaneous metadata files.
+#
+# We need to limit the files contained in the 'upgrade image" tarball
+# because the signing service that we use to sign the SHA256SUMS file
+# below, has a relatively small limit on the size of the file that we
+# can sign with it. Thus, we need the contents of the SHA256SUMS file to
+# be realtively small (and that file's size grows linearly with the
+# number of files contained in the "upgrade image" tarball).
+#
+tar -I pigz -cf "payload.tar.gz" -C upgrade-scripts . -C ~/.aptly .
+
 # Include version information about this image.
 VERSION=$(dpkg -f "$(find debs/ -name 'delphix-entire-*' | head -n 1)" version)
 sed "s/@@VERSION@@/$VERSION/" <version.info.template >version.info
 
-tar -I pigz -cf "$APPLIANCE_VARIANT.upgrade.tar.gz" version.info -C upgrade-scripts . -C ~/.aptly .
+sha256sum payload.tar.gz version.info >SHA256SUMS
+
+if [[ -n "${DELPHIX_SIGNATURE_TOKEN:-}" ]] && [[ -n "${DELPHIX_SIGNATURE_URL:-}" ]]; then
+	echo "{\"data\": \"$(base64 -w 0 SHA256SUMS)\"}" >sign-request.payload
+
+	#
+	# Here, we need to generate signature files for all of the appliance
+	# versions that'll be allowed to upgrade from, using this upgrade
+	# image. For now, we hardcode this to simply generate a signature for
+	# the "5.3" release, but eventually we'll need to update this code to
+	# support more versions (as more linux-based versions are released).
+	#
+	for signature_version in "5.3"; do
```

Comment:
Is there a reason not to port immediately to Linux the logic from `mkimage.sh` that obtains the list of all major releases by querying the app gate repo via `git ls-remote`?

Reply:
That's just work I haven't implemented yet, so I thought I could land this as-is and do that work afterwards. I'm also not sure if that implementation should change based on the linux build architecture changes. I want some more time to think about this, and about what the correct solution should be on linux, before porting over the illumos logic.